app/Controllers/userController.php

```diff
@@ -516,6 +528,42 @@ public function deleteAction() {
 		Minz_Request::forward($redirect_url, true);
 	}
 
+	public function promoteAction() {
+		$this->switchAdminAction(true);
+	}
+
+	public function demoteAction() {
+		$this->switchAdminAction(false);
+	}
+
+	private function switchAdminAction($isAdmin) {
+		if (!FreshRSS_Auth::hasAccess('admin')) {
+			Minz_Error::error(403);
+		}
+
+		if (!Minz_Request::isPost()) {
+			Minz_Error::error(403);
+		}
+
+		$username = Minz_Request::param('username');
+		if (!FreshRSS_UserDAO::exists($username)) {
+			Minz_Error::error(404);
+		}
+
+		$userConfig = get_user_configuration($username);
```

Comment:
```suggestion
		$userConfig = get_user_configuration($username);
		if ($userConfig == null) {
			Minz_Error::error(500);
		}
```

Reply:
`get_user_configuration()` might return null

Reply:
Done in a different way